rust/integration-tests/src/diff_tests/all_tests.rs

```diff
@@ -0,0 +1,181 @@
+#[allow(clippy::needless_return)]
+#[cfg(test)]
+mod test {
+
+    use crate::{
+        diff_test_helper::{
+            processors::{
+                event_processor::EventsProcessorTestHelper,
+                fungible_asset_processor::FungibleAssetProcessorTestHelper,
+                token_v2_processor::TokenV2ProcessorTestHelper,
+            },
+            ProcessorTestHelper,
+        },
+        diff_tests::{
+            get_expected_generated_txns, get_expected_imported_mainnet_txns,
+            get_expected_imported_testnet_txns, remove_inserted_at,
+        },
+        DiffTest, TestContext, TestProcessorConfig, TestType,
+    };
+    use aptos_indexer_test_transactions::{
+        ALL_GENERATED_TXNS, ALL_IMPORTED_MAINNET_TXNS, ALL_IMPORTED_TESTNET_TXNS,
+    };
+    use assert_json_diff::assert_json_eq;
+    use diesel::pg::PgConnection;
+    use processor::processors::token_v2_processor::TokenV2ProcessorConfig;
+    use std::{collections::HashMap, fs, sync::Arc};
+
+    #[tokio::test]
+    async fn test_all_testnet_txns_schema_output_for_all_processors() {
+        let processor_configs = get_processor_configs();
+        let test_context = TestContext::new(ALL_IMPORTED_TESTNET_TXNS).await.unwrap();
+
+        run_processor_tests(
+            processor_configs,
+            &test_context,
+            get_expected_imported_testnet_txns,
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn test_all_mainnet_txns_schema_output_for_all_processors() {
+        let processor_configs = get_processor_configs();
+        let test_context = TestContext::new(ALL_IMPORTED_MAINNET_TXNS).await.unwrap();
+
+        run_processor_tests(
+            processor_configs,
+            &test_context,
+            get_expected_imported_mainnet_txns,
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn test_all_generated_txns_schema_output_for_all_processors() {
+        let processor_configs = get_processor_configs();
+        let test_context = TestContext::new(ALL_GENERATED_TXNS).await.unwrap();
+
+        run_processor_tests(
+            processor_configs,
+            &test_context,
+            get_expected_generated_txns,
+        )
+        .await;
+    }
+
+    // Helper function to reduce duplicate code for running tests on all processors
+    async fn run_processor_tests<F>(
+        processor_configs: Vec<TestProcessorConfig>,
+        test_context: &TestContext,
+        get_expected_json_path_fn: F,
```

Comment:
is it due to the lifetime that you pass closure here instead of fn? 

```
get_expected_json_path_fn: fn(&str, &str) -> String
```

Reply:
actually, that should work too.